tm2/pkg/telemetry/traces/traces.go

```diff
@@ -0,0 +1,62 @@
+package traces
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/gnolang/gno/tm2/pkg/telemetry/config"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdkTrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+)
+
+func Init(cfg config.Config) (*sdkTrace.TracerProvider, error) {
+	var (
+		ctx = context.Background()
+		exp sdkTrace.SpanExporter
+	)
+
+	u, err := url.Parse(cfg.ExporterEndpoint)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing tracer exporter endpoint: %s, %w", cfg.ExporterEndpoint, err)
+	}
+
+	// Use oltp metric exporter with http/https or grpc
+	switch u.Scheme {
+	case "http", "https":
+		exp, err = otlptracehttp.New(
+			ctx,
+			otlptracehttp.WithEndpointURL(cfg.ExporterEndpoint),
+		)
+		if err != nil {
+			return nil, fmt.Errorf("unable to create http traces exporter, %w", err)
+		}
+	default:
+		exp, err = otlptracegrpc.New(
```

Comment:
We have a convention after talking to Jae, and that's that grpc fallbacks like this should be dropped entirely -- only support http / https